### ISSUE TITLE
Improve support for specifiers that include both a specimen identifier and a scientific name

### ISSUE THREAD

**Issue by gaurav** [phyloref/curation-tool#169]
Even when a specifier is a specimen, we usually also have information about the taxon or taxon concept that it had been identified to when the clade definition was created. It would be useful to record this information as well.

**Comment by gaurav** [comment]
While this is supported by the Phyx format, support for this feature has not yet been implemented in Klados.

**Comment by gaurav** [comment]
Depends on #232

**Comment by gaurav** [comment]
- [ ] Add a universal identifier (UUID) as a field

**Comment by gaurav** [comment]
Since implementing specimens with taxon name is no longer a priority (especially since specimens are rarely used in phyloreferences), I'm going to move this to enhancements not necessary for v1.0, and create a new issue just for adding UUIDs to specimens.
